# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/reftests/image-element/reftest.list
# Commit: 62d52e09851e
# Full Hash: 62d52e09851eeef665ca67ba97ee22139c174945
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2021-05-16 09:17:48
# Regressor Bug: 1540737
# File Overlap Count: 2
# Description:
#   Bug 1540737 - Implement Paint() for all display items, and nsDisplayList so that we can paint without using FrameLayerBuilder. r=miko
#   
#   This creates a new leaf nsDisplayWrapper item for consumers that just want an empty wrapper, with a Paint method that asserts. This lets us leaves Paint pure-virtual on nsDisplayWrapList, and lets the compiler force us to implement it for the other leaf types.
#   
#   It also disables flattening of transform/opacity items. This is because they flatten the items even when used with FlattenedDisplayItemIterator, even though this doesn't add markers to replace them (only FLB does this). We could in theory fix this, but I don't think FLB performance matters much any more.
# ==============================================================================

diff -r 00af531b9d94 -r 62d52e09851e layout/reftests/image-element/reftest.list
--- a/layout/reftests/image-element/reftest.list	Sat May 15 22:38:45 2021 +0000
+++ b/layout/reftests/image-element/reftest.list	Sat May 15 22:38:45 2021 +0000
@@ -25,9 +25,9 @@
 
 fails-if(usesRepeatResampling&&!(webrender&&winWidget)) == element-paint-subimage-sampling-restriction.html about:blank
 == element-paint-clippath.html element-paint-clippath-ref.html
-fuzzy-if(webrender,36-39,704-738) == element-paint-sharpness-01a.html element-paint-sharpness-01b.html
+fuzzy-if(!useDrawSnapshot,1-1,128-128) fuzzy-if(d2d,47-47,652-652) fuzzy-if(webrender,36-39,704-738) == element-paint-sharpness-01a.html element-paint-sharpness-01b.html
 fuzzy-if(skiaContent,0-1,0-326) fails-if(useDrawSnapshot) == element-paint-sharpness-01b.html element-paint-sharpness-01c.html
-fuzzy-if(webrender,36-39,704-738) == element-paint-sharpness-01c.html element-paint-sharpness-01d.html
+fuzzy-if(!useDrawSnapshot,1-1,128-128) fuzzy-if(d2d,47-47,652-652) fuzzy-if(webrender,36-39,704-738) == element-paint-sharpness-01c.html element-paint-sharpness-01d.html
 == element-paint-sharpness-02a.html element-paint-sharpness-02b.html
 fails-if(useDrawSnapshot) == element-paint-sharpness-02b.html element-paint-sharpness-02c.html
 == element-paint-paintserversize-rounding-01.html element-paint-paintserversize-rounding-01-ref.html